[AI-6411] Add HPE Aruba EdgeConnect integration (#23594)
DataDog/integrations-core: hpe_aruba_edgeconnect/README.md

- **QoS and traffic shaping**: Tracks per-DSCP-class bandwidth and rate counters, shaper drop counts, and drop percentages. Useful for validating QoS policies and spotting classes being starved or over-subscribed.
- **Circuit SLA probes**: Reports average latency, jitter, and packet loss from SLA probes, plus next-hop administrative and operational status, to monitor underlay link quality independently of the overlay.
- **Application performance**: Provides per-application latency, enabling drill-down from tunnel or interface issues to the specific affected applications.
- **Network Device Monitoring topology**: Pushes device, interface, and tunnel metadata to NDM, enabling visualization of the SD-WAN fabric and correlation with the rest of the network.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.

### Installation

The HPE Aruba EdgeConnect check is included in the [Datadog Agent][2] package.
No additional installation is needed on your server.

### Configuration

> 1. Edit the `hpe_aruba_edgeconnect.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your HPE Aruba EdgeConnect performance data. See the [sample hpe_aruba_edgeconnect.d/conf.yaml][4] for all available configuration options.

Reviewer: ```suggestion
1. In the `conf.d/` folder at the root of your Agent's configuration directory , edit the `hpe_aruba_edgeconnect.d/conf.yaml` file to start collecting your HPE Aruba EdgeConnect performance data. See the [sample hpe_aruba_edgeconnect.d/conf.yaml][4] for all available configuration options.
```
I find it helpful to list these things in the order the user will interact with them in; I think it makes it easier to follow 🙂

Author: This section of the README is the same across all integrations. Would it be better to follow the existing pattern, or do you think it's worth changing it?

Reviewer: Oh, no worries, it's really minor! I like to refer to things in the order that users will see them in because I think it's easier to follow, but it's not a hill to die on 🙂

- **Application performance**: Provides per-application latency, enabling drill-down from tunnel or interface issues to the specific affected applications.
- **Network Device Monitoring topology**: Pushes device, interface, and tunnel metadata to NDM, enabling visualization of the SD-WAN fabric and correlation with the rest of the network.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.

### Installation

The HPE Aruba EdgeConnect check is included in the [Datadog Agent][2] package.
No additional installation is needed on your server.

### Configuration

1. Edit the `hpe_aruba_edgeconnect.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your HPE Aruba EdgeConnect performance data. See the [sample hpe_aruba_edgeconnect.d/conf.yaml][4] for all available configuration options.

>    **Note**: Monitor permissions are enough to collect most metrics. Admin permissions are required to collect the `hpe_aruba_edgeconnect.device.cpu.usage` metric on the appliances. If the configured credentials do not have admin access, this metric will be skipped, but the rest of the metrics will still be collected.

Reviewer: ```suggestion
   **Note**: Monitor permissions are enough to collect most metrics. Admin permissions are required to collect the `hpe_aruba_edgeconnect.device.cpu.usage` metric on the appliances. If the configured credentials do not have admin access, the Agent skips this metric, but collects the rest of the metrics.
```
